File: erlang/euler1_1.erl
% Euler1 in Erlang

-module (euler1).
-export ([euler/0]).

euler() ->io:format( "~w~n", [euler(999, 0)] ).

euler(0, Acc) -> Acc;
euler(X, Acc) when X rem 3 == 0; X rem 5 == 0 ->
  euler(X-1, Acc+X);
euler(X, Acc) -> euler(X-1, Acc).

File: r/euler1_1.r
# Euler1 in R

#Set recursion limit
options(expressions = 10000)

myInt <- function(n) ifelse (n%%3 == 0 || n%%5 == 0, n, 0)

euler <- function(n, acc) {
    ifelse (n == 0,
        acc,
        euler(n-1, myInt(n)+acc))
}

euler1 <- function(n) euler(n, 0)

euler1(999)
